### PR TITLE
Remove setuptools dependency

### PR DESCRIPTION
**Pull request by DimedS**
## Description
Remove setuptools dependency as mentioned in #2350 

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by DimedS** [comment]
@astrojuanlu I saw your proposals:
https://github.com/kedro-org/kedro/issues/2350#issuecomment-1776155871
but it looks like they both are breaking compatibility changes?

on a first step currently I just emulated setuptools autodiscovery, but maybe it's not a robust solution.
Actually, if a question is just about package name, looks like we can take it from pyproject.toml: from section
```
[project]
name = 
```
but maybe it's not a mandatory that it must be pyproject.toml file there, and it will not be a check for quantity of packages also.

File: kedro/framework/cli/micropkg.py [hunk 1]
```diff
@@ -227,9 +226,11 @@ def _pull_package(  # noqa: PLR0913
         # However, the rest of the code expects the non-normalized package name,
         # so we have to find it.
         packages = [
-            package
-            for package in FlatLayoutPackageFinder().find(project_root_dir)
-            if "." not in package
+            elem.name
```
Comment (merelcht) [inline kedro/framework/cli/micropkg.py:229]: `elem` isn't a very descriptive variable name, can you rename it to be clearer? Something like project_item maybe?

**Review by merelcht — COMMENTED** [review]
Generally look good! 👍 Just left a minor comment. Also don't forget to update the release notes.

**Review by astrojuanlu — APPROVED** [review]
The logic looks correct to me 👍🏽

Indeed @DimedS my other proposals were not completely backwards compatible. And relying on `pyproject.toml` isn't either, because of all the transitions in 0.18 (from 1 pyproject.toml + 1 setup.py, to 2 pyproject.toml, to just 1 pyproject.toml)

Hopefully in the future we'll be able to simplify this logic (or even extract the packages directly from the metadata https://discuss.python.org/t/record-the-top-level-names-of-a-wheel-in-metadata/29494/69?u=astrojuanlu)

File: kedro/framework/cli/micropkg.py (hunk 1 above)
Comment (DimedS) [reply kedro/framework/cli/micropkg.py]: Thanks, fixed!

**Review by merelcht — APPROVED** [review]
LGTM! ⭐

**Pull request merged**